### ISSUE TITLE
Fullscreen doesn't work on Wayland backend

### ISSUE THREAD

**Issue by Zoarial94** [ValveSoftware/gamescope#1173]
TLDR: Passing the `-f` or `--fullscreen` arguments only provides a borderless window. Forcing the existing window into fullscreen causes gamescope to crash. This bug was intoduced with the new Wayland backend https://github.com/ValveSoftware/gamescope/commit/ce1156af053239e662aa021dbfc6a64cb770f64a.

Operating System: Fedora Linux 40
KDE Plasma Version: 6.0.0
KDE Frameworks Version: 6.0.0
Qt Version: 6.6.2
Kernel Version: 6.8.0-0.rc6.49.fc40.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 32 × AMD Ryzen 9 5950X 16-Core Processor
Memory: 125.7 GiB of RAM
Graphics Processor: AMD Radeon RX 6900 XT

I was attempting to run some HDR games, and ran into the same issue as #1161 using [the latest tag](https://github.com/ValveSoftware/gamescope/tree/3.14.2) from Fedora's repo. I decided to build from main like the issue suggested, but I couldn't get gamescope to run in fullscreen. Forcing the window into fullscreen would crash gamescope. After building from https://github.com/ValveSoftware/gamescope/commit/14a1db3a57612e5cfbba6d4c19688eafdc6c4043 I had no issues running fullscreen HDR content. 

The issue seems to be the Wayland backend. Changing main.cpp:755 to SDL instead of Wayland resolves the issue. 
https://github.com/ValveSoftware/gamescope/blob/e0c277a549320cd949b09d16008dab8fc6a9e73b/src/main.cpp#L752-L760

Running in borderless window using the Wayland backend seems to otherwise work fine. Zero Dawn Horizon and Doom Eternal both open and run.

**Comment by Zoarial94** [comment]
After some more debugging, I found the root cause. The program would crash with SIGFPE. The core dump revealed the crash occurred at wayland_backend.cpp:645. I'm assuming `uRefresh` is 0 and causing the crash when dividing by 0. 
https://github.com/ValveSoftware/gamescope/blob/e0c277a549320cd949b09d16008dab8fc6a9e73b/src/wayland_backend.cpp#L645

I surrounded the code with a simple check and I no longer had a crashes. I don't know the full context of this portion of the code, so I'm not assuming it is a good solution. 

I was able to run Doom Eternal without any issues `ENABLE_HDR_WSI=1 ~/src/gamescope/build/src/gamescope -f -W 3440 -H 1440 --hdr-enabled --hdr-debug-force-output --steam --debug-events -- env ENABLE_GAMESCOPE_WSI=1 DXVK_HDR=1 DISABLE_HDR_WSI=1 steam -bigpicture` and I used GLXGears for a quick and easy test. `~/src/gamescope/build/src/gamescope glxgears`. I can force the window in and out of fullscreen without any issues.

**Comment by Joshua-Ashton** [comment]
Yep, I have a fix for this locally, as well as -fullscreen implemented.

We get 0 on fullscreen on VRR because that's what the spec says, it's very strange and I disagree with it (and also we reeally need that info... bleh).

**Comment by Joshua-Ashton** [comment]
(As for why I haven't pushed it yet, I implemented while trying to get Helldivers to work in HDR then I got distracted playing Helldivers and forgot to clean up and push it :sweat_smile:)

**Comment by Joshua-Ashton** [comment]
Oh looks like you made a PR too. I'll merge our stuff together, mine has the WIN + F hotkey and some others implemented too. But I didn't realise you can just set a NULL output for the compositor to pick, neat.

**Comment by Zoarial94** [comment]
I'm excited to see what you have in store! I haven't spent much time actually trying to play a game with the Wayland backend, but it seems like it's pretty broken. I'm assuming you have many more things working locally. 

I see two weird issues currently. (Which might not exist once you push your changes)
* If a game's resolution is smaller than gamescope's, the mouse isn't able to reach the bottom and right side of the window. 
* If I alt-tab out, then click back in, my spacebar doesn't work until I click alt again.

**Comment by ninjaquick** [comment]
I wasn't able to use scrollwheel at all in Borderlands 3, same spacebar behavior after an alt+tab.

**Comment by Joshua-Ashton** [comment]
> If a game's resolution is smaller than gamescope's, the mouse isn't able to reach the bottom and right side of the window.

What compositor? Doesn't happen for me.

> If I alt-tab out, then click back in, my spacebar doesn't work until I click alt again.

That makes sense, I need to do more smarter stuff with modifiers and whatever.

> I wasn't able to use scrollwheel at all in Borderlands 3

I didn't implement anything scrollwheel yet =D

**Comment by Joshua-Ashton** [comment]
Should be fixed now

**Comment by Zoarial94** [comment]
> > If a game's resolution is smaller than gamescope's, the mouse isn't able to reach the bottom and right side of the window.
> 
> What compositor? Doesn't happen for me.
> 

KWin. I am unable to reproduce on the latest commit. 

> > If I alt-tab out, then click back in, my spacebar doesn't work until I click alt again.
> 
> That makes sense, I need to do more smarter stuff with modifiers and whatever.
> 

This seems to be an issue for all keys. Holding 'W' to move forward and Alt-Tabbing will cause my character to keep moving forward until I re-enter the window and click W.
